hugegraph-core/src/main/java/com/baidu/hugegraph/schema/PropertyKey.java

```diff
@@ -196,12 +196,14 @@ public String clazz() {
                 valid = this.checkDataType(value);
                 break;
             case SET:
-                valid = value instanceof Set;
-                valid = valid && this.checkDataType((Set<?>) value);
+                valid = value instanceof Collection;
+                valid = valid && this.checkDataType((Collection<?>) value);
+                valid = valid || this.checkDataType(value);
```

Comment:
return false if the value is not a collection

Reply:
in query case  , the value is not collection type

Reply:
if the value is not collection type, call checkDataType(toSet(value))